File: backend/auth-service/src/main/java/com/bank/auth/dto/RegisterRequest.java
package com.bank.auth.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

import java.util.Set;

public record RegisterRequest(
        @NotBlank(message = "Email обязателен")
        @Email(message = "Некорректный формат email")
        String email,

        @NotBlank(message = "Пароль обязателен")
        @Size(min = 6, message = "Пароль должен содержать минимум 6 символов")
        String password,

        @NotBlank(message = "Имя обязательно")
        String firstName,

        @NotBlank(message = "Фамилия обязательна")
        String lastName,

        String phone,

        Set<String> roles
) {}

File: ios/BankShared/Sources/BankShared/DTOs/Requests/RegisterRequest.swift
import Foundation

public struct RegisterRequest: Codable, Sendable {
    public let email: String
    public let password: String
    public let firstName: String
    public let lastName: String
    public let phone: String?
    public let roles: Set<String>?

    public init(email: String, password: String, firstName: String, lastName: String, phone: String? = nil, roles: Set<String>? = nil) {
        self.email = email
        self.password = password
        self.firstName = firstName
        self.lastName = lastName
        self.phone = phone
        self.roles = roles
    }
}
